Description: Fix for CVE-2025-29070
 lcms2 (2.14-2) unstable; urgency=medium
 .
   * New patch: allow-to-read-portions-of-tag.diff.
     Thanks to s3v <c0llapsed@yahoo.it> (Closes: #1028664)
   * Update copyright information for plugins.
     Thanks to Florian Ernst <florian_ernst@gmx.net>
   * Bump standards version to 4.6.2, no changes needed
Author: Thomas Weber <tweber@debian.org>
Bug-Debian: https://bugs.debian.org/1028664

---
The information above should follow the Patch Tagging Guidelines, please
checkout https://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin:  https://github.com/mm2/Little-CMS/commit/ec399d6879184e92a88c9099c60573f35e82e28b
Bug: <upstream-bugtracker-url>
Bug-Debian: https://security-tracker.debian.org/tracker/CVE-2025-29070
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: (no|not-needed|<patch-forwarded-url>)
Applied-Upstream: <version>, (<commit-url>|commit:<commid-id>)
Reviewed-By: <name and email of someone who approved/reviewed the patch>
Last-Update: 2025-05-28

--- lcms2-2.14.orig/src/cmsgamma.c
+++ lcms2-2.14/src/cmsgamma.c
@@ -1144,7 +1144,8 @@ cmsBool smooth2(cmsContext ContextID, cm
     cmsFloat32Number *c, *d, *e;
     cmsBool st;
 
-
+    if (m < 4 || lambda < MATRIX_DET_TOLERANCE) return FALSE;
+    
     c = (cmsFloat32Number*) _cmsCalloc(ContextID, MAX_NODES_IN_CURVE, sizeof(cmsFloat32Number));
     d = (cmsFloat32Number*) _cmsCalloc(ContextID, MAX_NODES_IN_CURVE, sizeof(cmsFloat32Number));
     e = (cmsFloat32Number*) _cmsCalloc(ContextID, MAX_NODES_IN_CURVE, sizeof(cmsFloat32Number));